pkg/labels/array.go

```diff
@@ -96,13 +96,41 @@ nextLabel:
 	return true
 }
 
+// Intersects returns true if ls contains at least one label in needed.
+// Because both arrays are sorted by key, this is O(max(N, M)) rather than
+// O(N*M)
+//
+// This has the same matching semantics as Has, namely,
+// ["k8s:foo=bar"].Intersects(["any:foo=bar"]) == true
+// ["any:foo=bar"].Intersects(["k8s:foo=bar"]) == false
+func (ls LabelArray) Intersects(needed LabelArray) bool {
+	n := 0
+	l := 0
+
+	// Both arrays are sorted; walk through until we get a match
+	for n < len(needed) && l < len(ls) {
+		if needed[n].Matches(&ls[l]) {
+			return true
+		}
+
+		// Not equal, bump
+		if needed[n].Key < ls[l].Key {
```

Comment:
I'm pretty sure this is **wrong wrong wrong** now, since sorting CIDR strings alphabetically does not necessarily match ordering. It was a cute trick, but a bad idea.

Reply:
Yup, totally wrong! `["cidr:11.11.11.11/32"].Intersects(["cidr:110.0.0.0/8", "cidr:8.0.0.0/5"])` should be true, but it wasn't.

Reply:
(fixed, but leaving for visibility)